src/core_modules/capture-core/metaDataStoreLoaders/IOUtils/quickStoreRecursively.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+import { quickStore } from './quickStore';
+import type {
+    ApiQuery,
+    StoreName,
+    ConvertFn,
+    QuickStoreRecursivelyOptions,
+} from './IOUtils.types';
+
+type ExecuteOptions = {
+    onConvert?: ConvertFn,
+};
+
+type Variables = {
+    iteration: number,
+};
+
+type RecursiveQuery = {
+    ...ApiQuery,
+    params?: (variables: Variables) => Object,
+};
+
+type QuickStoreIterationOptions = {
+    variables: Variables,
+    onConvert?: ConvertFn,
+};
+
+const quickStoreIteration = async (
+    recursiveQuery: RecursiveQuery,
+    storeName: StoreName,
+    {
+        variables,
+        onConvert,
+    }: QuickStoreIterationOptions,
+) => {
+    const { rawResponse } = await quickStore(recursiveQuery, storeName, {
+        onConvert,
+        variables,
+    });
+
+    return !(rawResponse && rawResponse.pager && rawResponse.pager.nextPage);
+};
+
+const executeRecursiveQuickStore = (
+    recursiveQuery: RecursiveQuery,
+    storeName: StoreName,
+    {
+        onConvert,
+    }: ExecuteOptions) => {
+    const next = async (iteration: number = 1) => {
+        const done = await quickStoreIteration(recursiveQuery, storeName, {
+            onConvert,
+            variables: {
+                iteration,
+            },
+        });
+
+        if (!done) {
+            await next(iteration + 1);
+        }
+    };
+
+    return next();
+};
+
+const getRecursiveQuery = (query: ApiQuery, iterationSize: number) => ({
+    ...query,
+    params: (variables: Variables) => ({
+        ...query.params,
+        pageSize: iterationSize,
+        page: variables.iteration,
+    }),
+});
+
+export const quickStoreRecursively = async (
+    query: ApiQuery,
+    storeName: StoreName,
+    {
+        iterationSize = 500,
```

Comment:
What made you choose `500`?

Reply:
Experience and gut feeling. If we want to do something more sophisticated, let's leave it for another issue.

Reply:
I was more curious here on what is going on.